src/common/histogram.cpp

```diff
@@ -0,0 +1,50 @@
+#include "histogram.hpp"
+
+namespace shar {
+
+Histogram::Histogram() 
+  : m_family(nullptr)
+  , m_histogram(nullptr)
+{}
+
+Histogram::Histogram(const MetricsContext& context, std::shared_ptr<prometheus::Registry> registry, 
+                                                                       std::vector<double>& bounds) {
+  std::map<std::string, std::string> labels = { {context.m_name, context.m_output_type} };
+  m_family = &prometheus::BuildHistogram()
+    .Name(context.m_name)
+    .Help(context.m_help)
+    .Labels(std::move(labels))
+    .Register(*registry);
+  m_histogram = &m_family->Add(labels, bounds);
+}
+
+Histogram::Histogram(Histogram&& histogram)
+  : m_histogram(histogram.m_histogram)
+  , m_family(histogram.m_family) {
+  histogram.m_family = nullptr;
+  histogram.m_histogram = nullptr;
+}
+
+Histogram& Histogram::operator=(Histogram&& histogram) {
+  if (this != &histogram) {
+    m_histogram = histogram.m_histogram;
+    m_family = histogram.m_family;
+    histogram.m_family = nullptr;
```

Comment:
Use `std::unique_ptr` with `MetricsFamilyRemover` as `Deleter`, so move constructor could be declared as `default`